doc/Programming-guide/Key-platform-features/Common-Type-Library/index.md

```diff
@@ -9,23 +9,24 @@ sort_idx: 20
 * TOC
 {:toc}
 
-The *Common Type Library (CTL)* is a repository of reusable data type schemas that can be referenced and reused as the user creates specific schema definitions for Kaa modules.
-The *Common Type (CT)* is a unit of CTL which represents set of *data type schema versions (CT schemas)*.
-This capability unifies management of all schemas in a Kaa server instance.
+The [Common type library (CTL)]({{root_url}}Glossary/#common-type-library-ctl) is a repository of data type schemas used for all Kaa modules.
+As more schema types and versions are created, they are recorded in the CTL for future use.
 
-# CT types definitions and FQNs
+[Common type (CT)]({{root_url}}Glossary/#common-type-ct) is a CTL unit representing a set of data type schema versions.
+Using CTL allows for consistent schema management within a [Kaa instance]({{root_url}}Glossary/#kaa-instance-kaa-deployment).
 
-CT is identified by FQN (Fully Qualified Name).
-FQN is combination of namespace and name attributes defined in root avro record of CT schema according to [Names in Avro  Schema Declaration](http://avro.apache.org/docs/current/spec.html#names).
-Each CT contains set of CT schemas with the same FQN and different versions.
-CT schema is identified by FQN and version which is unique across the CT.
-CT schema becomes unmodifiable after creation and only can be deleted.
+## CT definitions and FQNs
 
-# CT schema versioning and dependencies
+Every CT is identified by **Fully qualified name (FQN)**.
+FQN is a combination of namespace and name attributes defined in the root Avro record of a CT schema in compliance with [Apache Avro specification](http://avro.apache.org/docs/current/spec.html#names).
```

Comment:
It's not "in compliance", Kaa schema definition format is inherited from Avro, and mostly compatible with it. I suggest giving an example here, and then analysing it in subsequent sections step by step.